zeppelin-jupyter/src/main/java/org/apache/zeppelin/jupyter/JupyterUtil.java

```diff
@@ -0,0 +1,204 @@
+import org.apache.zeppelin.jupyter.nbformat.DisplayData;
+import org.apache.zeppelin.jupyter.nbformat.Error;
+import org.apache.zeppelin.jupyter.nbformat.ExecuteResult;
+import org.apache.zeppelin.jupyter.nbformat.HeadingCell;
+import org.apache.zeppelin.jupyter.nbformat.MarkdownCell;
+import org.apache.zeppelin.jupyter.nbformat.Nbformat;
+import org.apache.zeppelin.jupyter.nbformat.Output;
+import org.apache.zeppelin.jupyter.nbformat.RawCell;
+import org.apache.zeppelin.jupyter.nbformat.Stream;
+import org.apache.zeppelin.jupyter.zformat.Note;
+import org.apache.zeppelin.jupyter.zformat.Paragraph;
+import org.apache.zeppelin.jupyter.zformat.Result;
+import org.apache.zeppelin.jupyter.zformat.TypeData;
+
+/**
+ *
+ */
+public class JupyterUtil {
+
+  private static final String TEXT_PLAIN = "text/plain";
+  private static final String IMAGE_PNG = "image/png";
+
+  private final RuntimeTypeAdapterFactory<Cell> cellTypeFactory;
+  private final RuntimeTypeAdapterFactory<Output> outputTypeFactory;
+
+  public JupyterUtil() {
+    this.cellTypeFactory = RuntimeTypeAdapterFactory.of(Cell.class, "cell_type")
+        .registerSubtype(MarkdownCell.class, "markdown").registerSubtype(CodeCell.class, "code")
+        .registerSubtype(RawCell.class, "raw").registerSubtype(HeadingCell.class, "heading");
+    this.outputTypeFactory = RuntimeTypeAdapterFactory.of(Output.class, "output_type")
+        .registerSubtype(ExecuteResult.class, "execute_result")
+        .registerSubtype(DisplayData.class, "display_data").registerSubtype(Stream.class, "stream")
+        .registerSubtype(Error.class, "error");
+  }
+
+  public Nbformat getNbformat(Reader in) {
+    return getNbformat(in, new GsonBuilder());
+  }
+
+  public Nbformat getNbformat(Reader in, GsonBuilder gsonBuilder) {
+    return getGson(gsonBuilder).fromJson(in, Nbformat.class);
+  }
+
+  public Note getNote(Reader in, String codeReplaced, String markdownReplaced) {
+    return getNote(in, new GsonBuilder(), codeReplaced, markdownReplaced);
+  }
+
+  public Note getNote(Reader in, GsonBuilder gsonBuilder, String codeReplaced,
+      String markdownReplaced) {
+    return getNote(getNbformat(in, gsonBuilder), codeReplaced, markdownReplaced);
+  }
+
+  public Note getNote(Nbformat nbformat, String codeReplaced, String markdownReplaced) {
+    Note note = new Note();
+
+    String name = nbformat.getMetadata().getTitle();
+    if (null == name) {
+      name = "Note converted from Jupyter";
+    }
+    note.setName(name);
+
+    String lineSeparator = System.lineSeparator();
+    Paragraph paragraph;
+    List<Paragraph> paragraphs = new ArrayList<>();
+    String interpreterName;
+    List<TypeData> typeDataList;
+    String type;
+    String result;
+
+    for (Cell cell : nbformat.getCells()) {
+      paragraph = new Paragraph();
+      typeDataList = new ArrayList<>();
+
+      if (cell instanceof CodeCell) {
+        interpreterName = "%" + codeReplaced;
```

Comment:
In my opinion this ampersand should not be here.
If we import a note, we can either set a default interpreter or produce an incorrect query if the interpreter does not exist.
For example,
```
Note note = new JupyterUtil().getNote(in, "", "");
====
    {
      "text": "%\n",
      "results": {
        "code": "SUCCESS",
        "msg": []
      },
      "id": "20170609-193838_441716837",
      "status": "FINISHED"
    }
```

so in my opinion, It would be better to accept the interpreter raw tag from the beginning.
```
Note note = new JupyterUtil().getNote(in, "%python", "%md");
```